build.py

```diff
@@ -84,11 +92,21 @@ def setup_step():
 
 
 def build_step():
+    git_hash = subprocess.check_output(["git", "rev-parse", "HEAD"], cwd=LOCAL).strip().decode("utf-8")
+
+    print(f"Using git hash: {git_hash}")
+    with open(GIT_HASH_FILE, 'w') as f:
+        f.write(git_hash)
+
+    DATA_FILES.append(os.path.relpath(GIT_HASH_FILE, start=LOCAL))
+
     print("Running PyInstaller...")
     hidden_imports = [f"--hidden-import={i}" for i in HIDDEN_IMPORTS]
-    data_files = [f"--add-data={i}{PYINSTALLER_SEPARATOR}{os.path.dirname(i)}" for i in DATA_FILES]
```

Comment:
`os` provides methods to get the absolute directory with no symlinks, from which you can get an absolute path to the parent, but not sure if that's what you were looking for

Reply:
We dont need the absolute path, we need the relative path. Also yeah we use the `os` function `os.path.dirname()` for getting a parent dir in the `parent_dir()` function.

But the reason for the `parent_dir()` wrapper around `os.path.dirname(path)` is because `os` returns `''` as the parent dir to `'dir/'`, but pyinstaller expects `'.'`